stack/create_test.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stack_test
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/madlambda/spells/assert"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/stack"
+	"github.com/mineiros-io/terramate/test"
+	"github.com/mineiros-io/terramate/test/sandbox"
+)
+
+func TestStackCreation(t *testing.T) {
+	type wantedStack struct {
+		id      hcl.StackID
+		name    string
+		desc    string
+		imports []string
+	}
+	type want struct {
+		err   error
+		stack wantedStack
+	}
+	type testcase struct {
+		name   string
+		layout []string
+		create stack.CreateCfg
```

Comment:
just an idea:

we can have a `create` package inside `cmd/terramate/cli` and then here we would use `create.StackConfig` for the `terramate create`.

feel free to ignore.

Reply:
hmm I'm really not seeing why this here is cli specific :thinking:.